ui/cap-react/src/antd/forms/widgets/MaskedInput/MaskedInput.js

```diff
@@ -1,313 +1,69 @@
-    //             return true;
-    //           }
-    //           return false;
-    //         });
-    //         if (maskIndex === undefined && valueParts.length < mask.length) {
-    //           maskIndex = valueParts.length; // first unused one
-    //         }
-    //         if (maskIndex && mask[maskIndex].fixed) {
-    //           maskIndex -= 1; // fixed mask parts are never "active"
-    //         }
-    //         if (maskIndex !== activeMaskIndex) {
-    //           setActiveMaskIndex(maskIndex);
-    //         }
-    //       }, 10); // 10ms empirically chosen
-    //       return () => clearTimeout(timer);
-    //     }
-    //     return undefined;
-    //   },
-    //   [activeMaskIndex, focus, inputRef, mask, valueParts]
-    // );
-
-    // const setInputValue = useCallback(
-    //   nextValue => {
-    //     // Calling set value function directly on input because React library
-    //     // overrides setter `event.target.value =` and loses original event
-    //     // target fidelity.
-    //     // https://stackoverflow.com/a/46012210 &&
-    //     const nativeInputValueSetter = Object.getOwnPropertyDescriptor(
-    //       window.HTMLInputElement.prototype,
-    //       "value"
-    //     ).set;
-    //     nativeInputValueSetter.call(inputRef.current.input, nextValue);
-
-    //     const event = new Event("input", { bubbles: true });
-    //     inputRef.current.input.dispatchEvent(event);
-    //   },
-    //   [inputRef]
-    // );
-
-    // // This could be due to a paste or as the user is typing.
-    // const onChangeInput = useCallback(
-    //   event => {
-    //     // Align with the mask.
-
-    //     const nextValueParts = parseValue(mask, event.target.value);
-    //     const nextValue = nextValueParts.map(part => part.part).join("");
-
-    //     if (nextValue !== event.target.value) {
-    //       // The mask required inserting something, change the input.
-    //       // This will re-trigger this callback with the next value
-    //       setInputValue(nextValue);
-    //     } else if (value !== nextValue) {
-    //       setValue(nextValue);
-    //       if (onChange) onChange(event);
-    //     }
-    //   },
-    //   [mask, onChange, setInputValue, setValue, value]
-    // );
-
-    const renderPlaceholder = () => {
-      return mask.map((item) => item.placeholder || item.fixed).join("");
-    };
-
-    // const status = new RegExp(schemaMask).test(value);
-
-    return (
-      <div>
+import InputMask from "react-input-mask";
+
+const MAPPINGS = {
+  a: /[a-z]/,
+  A: /[A-Z]/,
+  "0": /[0-9]/,
+  "*": /[a-zA-Z0-9]/,
+};
+
+const MaskedInput = ({
+  id,
+  mask,
+  pattern,
+  name,
+  onBlur,
+  onChange,
+  onFocus,
+  onPressEnter,
+  placeholder,
+  buttons,
+  value,
+  disabled,
+  message,
+}) => {
+  const status = new RegExp(pattern).test(value);
+
+  return (
+    <div>
+      <InputMask
+        mask={
+          mask &&
+          mask
```

Comment:
The regex could probably be improved in order to avoid the filter afterwards (which is needed since the list contains some empty entries), but it works.